CMakeLists.txt

```diff
@@ -13,20 +13,19 @@ mark_as_advanced(VERSION_MAJOR VERSION_MINOR VERSION_PATCH COSE_VERSION)
 
 project ("cose-c" VERSION "${COSE_VERSION}")
 
-find_package(Doxygen)
-find_package(OpenSSL REQUIRED)
-
 ### setup options
 option (use_context	"Use context pointer for COSE functions" ON)
 option (verbose         "Produce verbose makefile output" OFF)
 option (optimize        "Optimize for size" OFF)
 option (fatal_warnings  "Treat build warnings as error" OFF)
 option (coveralls       "Generate coveralls data" ON)
 option ( coveralls_send "Send data to coveralls site" OFF )
-option (build_docs      "Create docs using Doxygen" ${DOXYGEN_FOUND} )
-option (build_tests      "build tests" ON )
-option (build_shared_libs "Build Shared Libraries" ON)
-option (use_embedtls    "Use MBedTLS for the Crypto Package" OFF)
+option (COSE_C_BUILD_DOCS      "Create docs using Doxygen" OFF )
```

Comment:
when using add_subdirectory workflow having different packages using the same variables causes lots of issues... e.g. cn-cbor uses the same `optimize`, `coveralls` etc.

therefore the de facto standard in cmake is use variables ala c-style macros prefixed with a pseudo namespace

cleaning up the rest would happen in a next MR